rust/src/dataset.rs

```diff
@@ -443,13 +443,15 @@ impl Dataset {
         let latest_manifest = latest_manifest_path(&base);
         let mut indices = vec![];
         let mut version = 1;
+        let mut dataset_fragments: Vec<Fragment> = Vec::new();
         let schema = if object_store.exists(&latest_manifest).await? {
             let dataset = Self::open(base_uri).await?;
             version = dataset.version().version + 1;
 
             if matches!(mode, WriteMode::Append) {
                 // Append mode: inherit indices from previous version.
                 indices = dataset.load_indices().await?;
+                dataset_fragments = dataset.fragments().iter().map(|f| f.clone()).collect();
```

Comment:
This will create duplicate paths in the current write append code path. The old fragments are already added here:

https://github.com/lancedb/lance/blob/d3e81539ee90a8c73d5c171a400d22248311737f/rust/src/dataset.rs#L355-L358

This is a breaking change in behavior of a public API, so I think I'd rather not make this change here if we can avoid it. But if we do make this change, we need to adjust the other code path as well, and add tests to make sure we aren't creating duplicate fragment entries.